### ISSUE TITLE
app.pushDelta error

### ISSUE THREAD

**Issue by inspirity** [joabakk/signalk-polar#68]
The plugin can't send any delta. Signalk-server Version 1.40 as Docker container with Debian 10.
Several data paths stop working. like environment.wind.speedApparent

it seems the app.pushDelta should be pushDelta without app. since pushDelta is not an app prop?

When I start the plugin I get this error in the server log:

Sep 25 10:58:17 TypeError: app.pushDelta is not a function at update.values.reduce (/home/node/.signalk/node_modules/signalk-polar/index.js:531:25) at Array.reduce (<anonymous>) at delta.updates.forEach.update (/home/node/.signalk/node_modules/signalk-polar/index.js:408:38) at Array.forEach (<anonymous>) at FullSignalK.handleDelta (/home/node/.signalk/node_modules/signalk-polar/index.js:402:21) at FullSignalK.emit (events.js:203:15) at FullSignalK.addDelta (/home/node/signalk/node_modules/@signalk/signalk-schema/dist/fullsignalk.js:52:8) at doProcess (/home/node/signalk/lib/deltachain.js:10:13) at msg (/home/node/signalk/lib/deltachain.js:29:17) at Array.unsubscribes.push.app.registerDeltaInputHandler (/home/node/.signalk/node_modules/@signalk/calibration/index.js:82:15) at doProcess (/home/node/signalk/lib/deltachain.js:13:21) at DeltaChain.process (/home/node/signalk/lib/deltachain.js:6:16) at Function.Server.app.handleMessage (/home/node/signalk/lib/index.js:174:28) at Simple.result.pipeElements.(anonymous function).on.msg (/home/node/signalk/lib/pipedproviders.js:58:17) at Simple.emit (events.js:203:15) at addChunk (_stream_readable.js:288:12) at readableAddChunk (_stream_readable.js:269:11) at Simple.Readable.push (_stream_readable.js:224:10) at Simple.Transform.push (_stream_transform.js:151:32) at Simple._transform (/home/node/signalk/node_modules/@signalk/streams/simple.js:87:8) at Simple.Transform._read (_stream_transform.js:190:10) at Simple.Transform._write (_stream_transform.js:178:12) at doWrite (_stream_writable.js:415:12) at writeOrBuffer (_stream_writable.js:399:5) at Simple.Writable.write (_stream_writable.js:299:11) at ToSignalK.ondata (_stream_readable.js:710:20) at ToSignalK.emit (events.js:198:13) at addChunk (_stream_readable.js:288:12) at readableAddChunk (_stream_readable.js:269:11) at ToSignalK.Readable.push (_stream_readable.js:224:10) at ToSignalK.Transform.push (_stream_transform.js:151:32) at ToSignalK._transform (/home/node/signalk/node_modules/@signalk/streams/n2k-signalk.js:160:12) at ToSignalK.Transform._read (_stream_transform.js:190:10) at ToSignalK.Transform._write (_stream_transform.js:178:12)
Sep 25 10:58:17 2021-09-25T10:58:17.205Z signalk-polar 2014-08-15T19:00:58.071Z navigation.courseOverGroundTrue 3.2306
Sep 25 10:58:17 2021-09-25T10:58:17.206Z signalk-polar timediff 1.4869999885559082 , engine running? false stable? true last store 1
Sep 25 10:58:17 2021-09-25T10:58:17.207Z signalk-polar sailing
Sep 25 10:58:17 TypeError: app.pushDelta is not a function at update.values.reduce (/home/node/.signalk/node_modules/signalk-polar/index.js:531:25) at Array.reduce (<anonymous>) at delta.updates.forEach.update (/home/node/.signalk/node_modules/signalk-polar/index.js:408:38) at Array.forEach (<anonymous>) at FullSignalK.handleDelta (/home/node/.signalk/node_modules/signalk-polar/index.js:402:21) at FullSignalK.emit (events.js:203:15) at FullSignalK.addDelta (/home/node/signalk/node_modules/@signalk/signalk-schema/dist/fullsignalk.js:52:8) at doProcess (/home/node/signalk/lib/deltachain.js:10:13) at msg (/home/node/signalk/lib/deltachain.js:29:17) at Array.unsubscribes.push.app.registerDeltaInputHandler (/home/node/.signalk/node_modules/@signalk/calibration/index.js:82:15) at doProcess (/home/node/signalk/lib/deltachain.js:13:21) at DeltaChain.process (/home/node/signalk/lib/deltachain.js:6:16) at Function.Server.app.handleMessage (/home/node/signalk/lib/index.js:174:28) at Simple.result.pipeElements.(anonymous function).on.msg (/home/node/signalk/lib/pipedproviders.js:58:17) at Simple.emit (events.js:203:15) at addChunk (_stream_readable.js:288:12) at readableAddChunk (_stream_readable.js:269:11) at Simple.Readable.push (_stream_readable.js:224:10) at Simple.Transform.push (_stream_transform.js:151:32) at Simple._transform (/home/node/signalk/node_modules/@signalk/streams/simple.js:87:8) at Simple.Transform._read (_stream_transform.js:190:10) at Simple.Transform._write (_stream_transform.js:178:12) at doWrite (_stream_writable.js:415:12) at writeOrBuffer (_stream_writable.js:399:5) at Simple.Writable.write (_stream_writable.js:299:11) at ToSignalK.ondata (_stream_readable.js:710:20) at ToSignalK.emit (events.js:198:13) at addChunk (_stream_readable.js:288:12) at readableAddChunk (_stream_readable.js:269:11) at ToSignalK.Readable.push (_stream_readable.js:224:10) at ToSignalK.Transform.push (_stream_transform.js:151:32) at ToSignalK._transform (/home/node/signalk/node_modules/@signalk/streams/n2k-signalk.js:160:12) at ToSignalK.Transform._read (_stream_transform.js:190:10) at ToSignalK.Transform._write (_stream_transform.js:178:12)
Sep 25 10:58:17 2021-09-25T10:58:17.704Z signalk-polar 2014-08-15T19:00:58.520Z environment.wind.speedApparent 5.56
Sep 25 10:58:17 2021-09-25T10:58:17.706Z signalk-polar timediff 1.935999870300293 , engine running? false stable? true last store 1
Sep 25 10:58:17 2021-09-25T10:58:17.706Z signalk-polar sailing

**Comment by joabakk** [comment]
Thank you!
I'll get back to this, just adding some info as note to self 
from Slack: 
  tkurki:could be as simple as replacing app.pushDelta with app.handleMessage
  inspirity: I tried to replace the app.pushDelta too. I had to go back to version 2.4.4 to have it working again.

**Comment by pcvonz** [comment]
I think I fixed this by:

1. replacing `app.pushDelta` with `pushDelta`. Why is this function not referenced anywhere? https://github.com/joabakk/signalk-polar/blob/aaa8c400f5e2cbd554a630ae06b04963263f89b3/index.js#L1203
2. That led to another error, `plugin` in the body of `pushDelta` wasn't defined. But it is here: https://github.com/joabakk/signalk-polar/blob/aaa8c400f5e2cbd554a630ae06b04963263f89b3/index.js#L71
So I moved the function declaration into the body of the function that contained `plugin`.

And now it seems to be working 🤔
